packages/react-components/src/components/MeetingReactionOverlay.tsx

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/* @conditional-compile-remove(reaction) */
+import {
+  OverlayModeTypes,
+  Reaction,
+  ReactionResources,
+  VideoGalleryLocalParticipant,
+  VideoGalleryRemoteParticipant
+} from '../types';
+/* @conditional-compile-remove(reaction) */
+import React, { useLayoutEffect, useRef, useState } from 'react';
+/* @conditional-compile-remove(reaction) */
+import { ParticipantVideoTileOverlay } from './VideoGallery/ParticipantVideoTileOverlay';
+/* @conditional-compile-remove(reaction) */
+import { RemoteContentShareReactionOverlay } from './VideoGallery/RemoteContentShareReactionOverlay';
+
+/* @conditional-compile-remove(reaction) */
+/**
+ * Reaction overlay component props
+ *
+ * Can be used with {@link VideoTile}.
+ *
+ * @internal
+ */
+export interface MeetingReactionOverlayProps {
+  /**
+   * Reaction rendering overlay type. i.e. single reaction rendering on grid-tile, screen share mode overlay
+   */
+  overlayMode: OverlayModeTypes;
+  /**
+   * Received reaction when overlay mode is grid-tile
+   */
+  reaction?: Reaction;
+  /**
+   * Reactions resources' url and metadata.
+   */
+  reactionResources: ReactionResources;
+  /**
+   * Local participant's reaction event that comes from participant object.
+   */
+  localParticipant?: VideoGalleryLocalParticipant;
+  /**
+   * Remote participant's reaction event.
+   */
+  remoteParticipants?: VideoGalleryRemoteParticipant[];
+}
+
+/* @conditional-compile-remove(reaction) */
+/**
+ * Emoji max size
+ * @private
+ */
+const DEFAULT_EMOJI_MAX_SIZE_PX = 100;
+/* @conditional-compile-remove(reaction) */
+/**
+ * Emoji min size
+ * @private
+ */
+const DEFAULT_EMOJI_MIN_SIZE_PX = 32;
+
+/* @conditional-compile-remove(reaction) */
+/**
+ * Reaction overlay component
+ *
+ * Can be used with {@link VideoTile}.
+ *
+ * @internal
+ */
+export const MeetingReactionOverlay = (props: MeetingReactionOverlayProps): JSX.Element => {
+  const { overlayMode, reaction, reactionResources, localParticipant, remoteParticipants } = props;
+  const [emojiSize, setEmojiSize] = useState(1);
```

Comment:
For `sizes` for try to use the unit in the name because its ambiguous. Either that or a comment
```suggestion
  const [emojiSizePx, setEmojiSize] = useState(1);
```
```suggestion
  // Size in pixels of the reaction animation
  const [emojiSize, setEmojiSize] = useState(1);
```

Reply:
Why `1` here?